lib/view_component/instrumentation.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ViewComponent # :nodoc:
+  module Instrumentation
+    class << self
+      def included(mod)
+        mod.prepend(self)
+      end
+
+      def enabled?
+        Rails.application.config.view_component.instrumentation_enabled
+      end
+    end
+
+    def render_in(view_context, &block)
+      return super(view_context, &block) unless Instrumentation.enabled?
+
+      ActiveSupport::Notifications.instrument("render.view_component", identifier: self.class.short_identifier) do
```

Comment:
```suggestion
      ActiveSupport::Notifications.instrument("!render.view_component", identifier: self.class.short_identifier) do
```

I think we should follow the Rails convention and prefix this with a `!` to help signal that this shouldn't be used in production.